test/jdk/javax/swing/JComboBox/JComboBoxPopupMenuEventTest.java

```diff
@@ -0,0 +1,205 @@
+                if (!popupMenuVisibleLatch.await(3, TimeUnit.SECONDS)) {
+                    throw new RuntimeException(
+                            "Waited too long, but popupMenuWillBecomeVisible " +
+                            "not yet got called for " + laf);
+                }
+
+                hitKeys(KeyEvent.VK_ENTER);
+
+                // Verifying whether popupMenuWillBecomeInvisible method of
+                // PopupMenuListener gets called when popup menu disappears.
+                if (!popupMenuInvisibleLatch.await(3, TimeUnit.SECONDS)) {
+                    throw new RuntimeException(
+                            "Waited too long, but popupMenuWillBecomeInvisible" +
+                                               "not yet got called for " + laf);
+                }
+
+                System.out.println("Test passed for " + laf);
+            } finally {
+                SwingUtilities.invokeAndWait(
+                        JComboBoxPopupMenuEventTest::disposeFrame);
+            }
+        }
+    }
+
+    private static void mouseClick(JComponent jComponent) throws Exception {
+        final Point location = getLocationOnScreen(jComponent);
+        robot.mouseMove(location.x + 8, location.y + 8);
+        robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+        robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+    }
+
+    private static Point getLocationOnScreen(JComponent jComponent)
+            throws Exception {
+        final AtomicReference<Point> loc = new AtomicReference<>();
+        SwingUtilities
+                .invokeAndWait(() -> loc.set(jComponent.getLocationOnScreen()));
+        final Point location = loc.get();
+        return location;
+    }
+
+    private static void hitKeys(int... keys) {
+        for (int key : keys) {
+            robot.keyPress(key);
+        }
+
+        for (int i = keys.length - 1; i >= 0; i--) {
+            robot.keyRelease(keys[i]);
+        }
+    }
+
+    public static void createUI() {
+        frame = new JFrame();
+        JPanel panel = new JPanel();
+        searchTextField = new JTextField(6);
+        panel.add(searchTextField);
+        comboBox = new JComboBox(compStrs);
+        panel.add(comboBox);
+        comboBox.addPopupMenuListener(new PopupMenuListener() {
+            public void popupMenuWillBecomeVisible(PopupMenuEvent e) {
+                System.out.println("popupMenuWillBecomeVisible() got called");
+                popupMenuVisibleLatch.countDown();
+                comboBox.removeAllItems();
+                String text = searchTextField.getText().trim();
+                List<String> comps = Arrays.stream(compStrs)
+                                           .filter(compStr -> compStr
+                                                   .startsWith(text))
+                                           .collect(Collectors.toList());
+                for (String comp : comps) {
+                    comboBox.addItem(comp);
+                }
```

Comment:
Just a suggestion: use `.forEach` and add items to the `comboBox` right in the stream without creating an intermediate list.